test.js

```diff
@@ -1538,6 +1538,29 @@ function runTests(baseopts) {
             done();
           });
       });
+      it('should emit an unlink event when a file is updated and deleted just after that', function(done) {
+        var spy = sinon.spy();
+        var testPath = getFixturePath('subdir/add.txt');
+        var filename = sysPath.basename(testPath);
+        options.cwd = sysPath.dirname(testPath);
+        fs.mkdir(options.cwd, w(function() {
+          fs.writeFile(testPath, 'hello', w(function() {
+            stdWatcher()
+              .on('all', spy)
+              .on('ready', function() {
+                fs.writeFile(testPath, 'edit', simpleCb);
+                w(function() {
```

Comment:
You're using the `w()` wrapper and specifying your own timeout at 10ms to override the default. I'm suggesting to just leave it with the default as long as that works.

```js
                fs.writeFile(testPath, 'edit', w(function() {
                  fs.unlink(testPath, simpleCb);
                  waitFor([spy.withArgs('unlink')], function() {
                    spy.should.not.have.been.calledWith('change', filename);
                    spy.should.have.been.calledWith('unlink', filename);
                    done();
                  });
                }));
```